src/uct/ib/base/ib_iface.c

```diff
@@ -572,6 +576,34 @@ uct_ib_iface_res_domain_init(uct_ib_iface_res_domain_t *res_domain,
                   uct_ib_device_name(dev));
         return UCS_ERR_IO_ERROR;
     }
+#elif HAVE_DECL_IBV_ALLOC_TD
+    uct_ib_device_t *dev = uct_ib_iface_device(iface);
+    uct_ib_md_t     *md  = uct_ib_iface_md(iface);
+    struct ibv_parent_domain_init_attr attr;
+    struct ibv_td_init_attr td_attr;
+
+    if (iface->super.worker->thread_mode == UCS_THREAD_MODE_MULTI) {
+        td_attr.comp_mask = 0;
+        res_domain->td = ibv_alloc_td(dev->ibv_context, &td_attr);
+        if (res_domain->td == NULL) {
+            ucs_error("ibv_alloc_td() on %s failed: %m",
+                      uct_ib_device_name(dev));
+            return UCS_ERR_IO_ERROR;
```

Comment:
need to call `uct_ib_iface_res_domain_cleanup`, right?

Reply:
Yes, this cleanup I've added: 
https://github.com/openucx/ucx/pull/2743/commits/053cd02e04a1fbb0ddd59cb688b4dd974e1e415f#diff-694c9dba586b3d5c9d63a14795983496R607